code/game/objects/items/weapons/caution.dm

```diff
@@ -24,7 +24,7 @@
 /obj/item/caution/proximity_sign/attack_self(mob/user as mob)
 	if(ishuman(user))
 		var/mob/living/carbon/human/H = user
-		if(H.mind.assigned_role != "Janitor")
+		if(!(H.mind.has_antag_datum(/datum/antagonist/traitor) || H.mind.has_antag_datum(/datum/antagonist/mindslave) && !(H.mind.has_antag_datum(/datum/antagonist/mindslave/thrall))))
```

Comment:
Ah I see now. It should actually be:
```suggestion
		if(!H.mind.has_antag_datum(/datum/antagonist/traitor) && !ismindslave(H))
```
If user is not a traitor AND not a mindslave, return. They can't use it. But if they're one or the other, they can.

Reply:
That is a much simpler way to do it! Thanks again!